### DISCUSSION TITLE
Question on RetryCallback return value

### DISCUSSION THREAD

**Discussion by mqayyuum** [nodejs/undici#3824] [Q&A] [answered]
What should be the return value for RetryCallback function in the RetryAgent? 

The [documentation](https://undici.nodejs.org/#/docs/api/RetryAgent?id=parameter-retryhandleroptions) mentioned that it should return `void`, but the [type definition](https://github.com/nodejs/undici/blob/f98fbef4a990ec74c871ddbd3454472ee0355670/types/retry-handler.d.ts#L26-L35) of RetryCallback indicates `number` or `null`. If the latter is true, how the return value is used?

**Comment by metcoder95** [comment, marked as answer]
The type definition is wrong, the `RetryCallback` should return `void`.

Thanks for the heads up, would you like to send a PR to address it?

**Reply by mqayyuum** [reply]
Sure, I will come up with a PR soon. Thanks for confirming it
